plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/WebProjectUtil.java

```diff
@@ -38,10 +38,15 @@
  * Utility classes for processing WTP Web Projects (jst.web and jst.utility).
  */
 public class WebProjectUtil {
-  final static String DEFAULT_WEB_PATH = "src/main/webapp";
+  /** Default top-level locations for WEB_INF */
+  public static final String DEFAULT_WEB_PATH = "src/main/webapp";
+
+  /** All possible top-level locations for WEB_INF */
+  final static String[] DEFAULT_WEB_PATHS = {DEFAULT_WEB_PATH, "WebContent", "war", "web"};
```

Comment:
It's `final`, but anyone can actually replace elements of the array, so it is not immutable.